<a href="https://colab.research.google.com/github/royanagilnugroho/Python_for_Data_Science/blob/main/tugas_8_agil_nugroho77.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [8]:
# Membuat dataframes dari dictionary
import pandas as pd
import numpy as np

df = pd.DataFrame({'tipe_int': [510, 211], 'tipe_string': ['a', 'b']})
print(df)


   tipe_int tipe_string
0       510           a
1       211           b


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [7]:
import csv

# Membaca data dan memasukannya kedalam suatu variabel
df_from_csv = pd.read_csv('/content/diamonds.csv', sep=',')

print (df_from_csv)

       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
#Perbedaan Series dengan List
jika print series akan memunculkan index, sedangkan list tidak disertai index. Selain integer, untuk index pada series dapat menggunakan data string, misalnya index (a,b,c).
#Perbedaan Series dengan Dictionary
series mempunyai key yang dapat diduplikasi, atau berduplikat. Selain itu, series memiliki kemampuan slicing data. Sedangkan dictionary tidak dapat dislicing datanya dan key pada dictionary hanya akan terbaca salah satu jika terdapat key yang duplikat.
- Apa perbedaan dataframe dan series
#
- Buatlah suatu data series

Jawab: 

In [9]:
# code here

import pandas as pd
import numpy as np

data=pd.Series(['a','b','c','d','e'])
np_1=np.array(['1','2','3','4','5'])
data_np=pd.Series(np_1)

print(data)
print(data_np)

0    a
1    b
2    c
3    d
4    e
dtype: object
0    1
1    2
2    3
3    4
4    5
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [35]:
# code here

a=df_from_csv['price'].head(1)



b=df_from_csv['color'].tail(1)
print(a)
print(b)

0    326
Name: price, dtype: int64
53939    D
Name: color, dtype: object


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab:
Statistik deskriptif adalah metode yang berhubungan dengan pengumpulan atau penyajian data bertujuan untuk memberikan informasi yang berguna dari data tersebut. Salah satu yang paling menonjol dalam statistik deskriptif adalah dalam bentuk grafik, tabel, diagram atau jenis visual lainnya. Selain itu, statistik deskriptif mengandung ukuran-ukuran statistik seperti mean, median, modus, dan lain-lain.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [41]:
# code here
df_from_csv.describe(include='all')



Unnamed: 0         carat    cut  color clarity         depth  \
count   53940.000000  53940.000000  53940  53940   53940  53940.000000   
unique           NaN           NaN      5      7       8           NaN   
top              NaN           NaN  Ideal      G     SI1           NaN   
freq             NaN           NaN  21551  11292   13065           NaN   
mean    26970.500000      0.797940    NaN    NaN     NaN     61.749405   
std     15571.281097      0.474011    NaN    NaN     NaN      1.432621   
min         1.000000      0.200000    NaN    NaN     NaN     43.000000   
25%     13485.750000      0.400000    NaN    NaN     NaN     61.000000   
50%     26970.500000      0.700000    NaN    NaN     NaN     61.800000   
75%     40455.250000      1.040000    NaN    NaN     NaN     62.500000   
max     53940.000000      5.010000    NaN    NaN     NaN     79.000000   

               table         price             x             y             z  
count   53940.000000  53940.000000  53940.000000  53940.000000  53940.000000  
unique           NaN           NaN           NaN           NaN           NaN  
top              NaN           NaN           NaN           NaN           NaN  
freq             NaN           NaN           NaN           NaN           NaN  
mean       57.457184   3932.799722      5.731157      5.734526      3.538734  
std         2.234491   3989.439738      1.121761      1.142135      0.705699  
min        43.000000    326.000000      0.000000      0.000000      0.000000  
25%        56.000000    950.000000      4.710000      4.720000      2.910000  
50%        57.000000   2401.000000      5.700000      5.710000      3.530000  
75%        59.000000   5324.250000      6.540000      6.540000      4.040000  
max        95.000000  18823.000000     10.740000     58.900000     31.800000

Jawab:
Mean of Column 'Price' is 3932.799722	
standard of deviation of column 'depth' is 1.432621	
maximum value of column 'carat' is 5.010000	